test(components/indicators): add unit test to check a11y compliance

Blackbaud-SandhyaRajasabeson opened the pull request:
add a unit test that checks that toBeAccessible() passes both when horizontal and vertical

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/blackbaud/skyux?pullRequest=755) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/blackbaud/skyux?pullRequest=755) before we can accept your contribution.<br/><hr/>**Chandler Gonzales** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

Blackbaud-PatrickOFriel commented:
I'll take a look at this tomorrow. When you get a chance, go ahead and sign the CLA please

jcgsville commented:
I believe that I've signed it. If it's not showing up on your end that I have, let me know

Blackbaud-PatrickOFriel commented:
Yep I see it now, thanks!

Blackbaud-PatrickOFriel commented on js/sky/src/grids/grids.js:
```diff
@@ -961,6 +961,10 @@
                                         $scope.paginationOptions.currentPage = 1;
                                     }
 
+                                    if (!$scope.paginationOptions.boundaryLinks) {
```
This if statement isn't needed because in this case $scope.paginationOptions.boundaryLinks will already be false.

jcgsville commented:
I'm a bit confused as to how that works. Without that if statement, the boundary links boolean gets set to false every time. But I guess pagination element is already evaluated before the initializePagination() function. Anyway, I made the change, and it works now.

Blackbaud-PatrickOFriel commented on js/sky/src/grids/grids.js:
```diff
@@ -961,6 +961,8 @@
                                         $scope.paginationOptions.currentPage = 1;
                                     }
 
+                                    $scope.paginationOptions.boundaryLinks = false;
```
You don't need this either. In JavaScript, if an item is undefined, then it evaluates to `false`. All you need to do is pass along $scope.paginationOptions.boundaryLinks from the user to the template which has the uib-pagination directive.

Blackbaud-PatrickOFriel replied:
The only reason the other paginationOptions are being set in this function is because they are being defaulted to a value other than false.

jcgsville commented:
Ahh gotcha. Thank you for your patience

Blackbaud-PatrickOFriel commented:
No problem

Blackbaud-PatrickOFriel commented:
Approved

Commit:
test(components/indicators): add unit test to check a11y compliance

add a unit test that checks that toBeAccessible() passes both when horizontal and vertical

blackbaud-sky-build-user commented:
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/755/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/755/storybooks/ag-grid-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/755/storybooks/indicators-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/755/storybooks/layout-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/755/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/755/storybooks/pages-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/755/storybooks/popovers-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/755/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/755/storybooks/tabs-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/755/storybooks/tiles-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/755/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/755/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/755/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/755/playground/)

Blackbaud-SandhyaRajasabeson committed:
Merge branch 'main' into keyinfo-axe-test

Blackbaud-SandhyaRajasabeson committed:
Merge branch 'main' into keyinfo-axe-test
